### PR TITLE
FBCM-5203 Move FFI Definitions to `Kafka.FFI.*` Modules

### PR DESCRIPTION
**Pull request by boygao1992**
## What does this pull request do?

Move FFI definitions and types related to `Kafka.FFI.*` modules.

### CODE REVIEW

File: src/Kafka/Admin.purs [hunk 1]
```diff
@@ -14,140 +16,60 @@ import Effect as Effect
...
   }
 
--- | https://github.com/tulios/kafkajs/blob/d8fd93e7ce8e4675e3bb9b13d7a1e55a1e0f6bbf/types/index.d.ts#L183
--- |
--- | Unsupported
--- | * `retry?: RetryOptions`
-type AdminConfigImpl =
-  {
-  }
-
```
Comment (davezuch) [inline src/Kafka/Admin.purs]: Perhaps not relevant to this PR, but do we need `AdminConfig` and `AdminConfigImpl` defined at all if it's going to have no value? Can't we just construct an empty JS object in the FFI wherever it's needed?

**Review by davezuch — APPROVED** [review]
Changes look fine, it's just moving definitions around so doesn't seem like there's much to look closely at. 

Can you touch on the motivation though? It seems like this is an attempt to separate the serialization layer from the domain. That is, the FFI stuff can be thought of as internal, while what's left in the main modules is the library's public API—is that right?

File: src/Kafka/Admin.purs (hunk 1 above)
Comment (boygao1992) [reply src/Kafka/Admin.purs]: `AdminConfigImpl` is there to document what could be supported when necessary but not supported yet. I guess we could remove `AdminConfig` for now.

**Comment by boygao1992** [comment]
Thanks for the review!

**Pull request merged**